core/src/main/java/com/radixdlt/consensus/bft/VertexStoreAdapter.java

```diff
@@ -109,29 +109,26 @@ public boolean tryRebuild(VertexStoreState vertexStoreState) {
     return result.isPresent();
   }
 
-  public void insertTimeoutCertificate(TimeoutCertificate timeoutCertificate) {
-    vertexStore.insertTimeoutCertificate(timeoutCertificate);
+  public boolean insertTimeoutCertificate(TimeoutCertificate timeoutCertificate) {
```

Comment:
Can we copy the java doc to the adaptor to explain what the boolean means?

By "ignored" I guess it means "ignored because it's not the highest"?

Reply:
I'd have to copy all the docs, for consistency. Not sure if it makes sense? It's quite clear that the adapter forwards it to the main impl.

Extended the comment with ignore reason :+1:

Reply:
> It's quite clear that the adapter forwards it to the main impl.

@LukasGasior1 I now had a quick look at this entire "adapter" and it does not do much adapting TBH.
OFC I have no idea on the motivation (or dev history of this class), but ATM it looks more like a decorator - namely, an `EventDispatchingVertexStore`, doesn't it?

Such decorator would just need a class-level javadoc (e.g. "A decorator dispatching events according to the results of the underlying `VertexStore`").

Then, the methods' javadocs (which David wants) would be auto-propagated from the interface.
(of course we would first have to move them from the `VertexStoreJavaImpl` to the `VertexStore` interface, which is the right move anyway)

(the only odd, non-decorator-ish method left would be `hasCommittedVertexOrRootAtOrAboveRound()`, which looks more like a convenience method, which is easiest to maintain as a `default` within the interface itself).

I find this mechanical change worthwhile, but easily delegated to another PR.

Reply:
yup, sounds good. The original reason to introduce the adapter was the handle the java-side events once vertex store is in rust (which hasn't happened yet)

Reply:
> Then, the methods' javadocs (which David wants) would be auto-propagated from the interface.

Not sure how that works? The Adapter itself doesn't implement VertexStore

Reply:
> The Adapter itself doesn't implement VertexStore

Yeah, and in https://github.com/radixdlt/babylon-node/pull/246#discussion_r1069367993 I postulate that it should, since it is more of an "event-dispatching VertexStore decorator" than adapter.

Since my Java is less ambiguous than my English, I imagined something like this:
```
public class EventDispatchingVertexStore implements VertexStore {
  private final VertexStore underlying;

  @Override
  public boolean tryRebuild(VertexStoreState vertexStoreState) {
    var result = this.underlying.tryRebuild();
    this.bftRebuildDispatcher.dispatch(whatever);
    return result;
  }
}
```

I agree that at this point the only thing it gives us is "contract becomes more obvious + javadoc gets propagated + less apis exist". Functionally it isn't a "must-have", just a minor refactor that could happen.

Reply:
Ah, gotcha. But currently VertexStoreAdapter's interface is slightly different. VertexStore was modified to make Java->Rust transition easier, while VertexStoreAdapter remained unchanged (i.e. it has the original VertexStore interface) to support existing clients. TBH, I wouldn't spend much time refining it in the current form, given its uncertain future.